### PR TITLE
More general handling of XSens exported txt files

### PR DESCRIPTION
**Pull request by aymanhab**
Fixes issue #2945 

### Brief summary of changes
Rearrange reader to treat comment lines as of arbitrary number but on the form // key:value to handle more recent formats

### CHANGELOG.md (choose one)

- no need to update because...
- updated.

---
This change is [Reviewable](https://reviewable.io/reviews/opensim-org/opensim-core/2947)

### CODE REVIEW

**Comment by aymanhab** [comment]
@antoinefalisse or @carmichaelong for a quick review

File: OpenSim/Common/XsensDataReader.cpp [hunk 1]
```diff
@@ -51,51 +52,46 @@ XsensDataReader::extendRead(const std::string& folderName) const {
...
-                    dataRate = std::stod(tokens[3]);
-                }
+        auto commentLine = true;
+        std::getline(*nextStream, line);
+        auto isCommentLine = [](std::string aline) {
+            return aline.substr(0, 2) == "//";
+        };
+        std::vector<std::string> tokens;
+        while (commentLine) {
+            // Comment lines of arbitrary naumber on the form // "key":"value"
```
Comment (carmichaelong) [inline OpenSim/Common/XsensDataReader.cpp:62]: Typo in this line (I think should be "Comment lines of arbitrary number in the form")

And then spacing in next line: `tokens = FileAdapter::tokenize(line.substr(2), ":");`

**Review by carmichaelong — COMMENTED** [review]
Code LGTM to be overall @aymanhab.

One thing I'd like to see additionally would be an additional test in `testXsensDataReader` that shows this working with a file that didn't work beforehand (assuming you might have one from the user?)

File: OpenSim/Common/XsensDataReader.cpp (hunk 1 above)
Comment (aymanhab) [reply OpenSim/Common/XsensDataReader.cpp:62]: 👍

**Comment by aymanhab** [comment]
Just got files to test with from our user, so will beef up the test case. Thanks for the review @carmichaelong

**Comment by aymanhab** [comment]
Test case beefed up. Ready for review.

**Comment by carmichaelong** [comment]
Thanks @aymanhab LGTM

**Review by carmichaelong — APPROVED** [review]

**Comment by aymanhab** [comment]
Thanks @carmichaelong

**Pull request merged**